spark/src/test/scala/org/apache/spark/sql/delta/DeltaTypeWideningSuite.scala

```diff
@@ -1252,6 +1399,50 @@ trait DeltaTypeWideningTableFeatureTests {
     assert(metrics("numFilesRewritten").toLong === 1L)
     assert(metrics("metadataRemoved").toBoolean)
   }
+
+  test("dropping feature after CLONE correctly rewrite files with old type") {
+    withTable("source") {
+      sql("CREATE TABLE source (a byte) USING delta")
+      sql("INSERT INTO source VALUES (1)")
+      sql("INSERT INTO source VALUES (2)")
+      sql(s"ALTER TABLE source CHANGE COLUMN a TYPE INT")
+      sql("INSERT INTO source VALUES (200)")
+      sql(s"CREATE OR REPLACE TABLE delta.`$tempPath` SHALLOW CLONE source")
+      checkAnswer(readDeltaTable(tempPath),
+        Seq(Row(1), Row(2), Row(200)))
+
+      dropTableFeature(
+        expectedOutcome = ExpectedOutcome.FAIL_CURRENT_VERSION_USES_FEATURE,
+        expectedNumFilesRewritten = 3,
+        expectedTypeAfterRewrite = IntegerType
+      )
+      checkAnswer(readDeltaTable(tempPath),
+        Seq(Row(1), Row(2), Row(200)))
+    }
+  }
+
+  test("dropping feature after RESTORE correctly rewrite files with old type") {
+    addSingleFile(Seq(1), ShortType)
+    addSingleFile(Seq(2), ShortType)
+    sql(s"ALTER TABLE delta.`$tempPath` CHANGE COLUMN a TYPE INT")
+    // Delete the first file which will be added back again with RESTORE.
+    sql(s"DELETE FROM delta.`$tempPath` WHERE a = 1")
+    addSingleFile(Seq(Int.MinValue), IntegerType)
+
+    // RESTORE to version 2 -> ALTER TABLE CHANGE COLUMN TYPE.
+    // The type change is then still present and the first file initially added at version 0 is
```

Comment:
What will happen when restoring to before the type change? We should still require history truncation in this case.

Reply:
I added a test to cover this. Dropping the feature will indeed fail as historical versions still contain the feature, asking to wait for the retention period to expire